edsnlp/pipelines/core/contextual_matcher/contextual_matcher.py

```diff
@@ -0,0 +1,446 @@
+import re
+from functools import lru_cache
+from typing import Any, Dict, List, Tuple, Union
+
+from loguru import logger
+from spacy.language import Language
+from spacy.tokens import Doc, Span
+
+from edsnlp.matchers.phrase import EDSPhraseMatcher
+from edsnlp.matchers.regex import RegexMatcher, create_span
+from edsnlp.matchers.utils import get_text
+from edsnlp.pipelines.base import BaseComponent
+from edsnlp.utils.filter import filter_spans
+
+from . import models
+
+
+@lru_cache(64)
+def get_window(
+    doclike: Union[Doc, Span],
+    window: int,
+    side: str,
+):
+    if side == "before":
+        snippet = doclike.doc[
+            max(doclike.start - window, doclike.sent.start) : doclike.end
+        ]
+    elif side == "after":
+        snippet = doclike.doc[
+            doclike.start : min(doclike.end + window, doclike.sent.end)
+        ]
+    return snippet
+
+
+class GroupRegexMatcher(RegexMatcher):
```

Comment:
Wouldn't it be better to add a parameter `group_idx` (or something similar) to the RegexMatcher class ?

Reply:
Yes good thinking, the option was added under the `span_from_group` argument in the `RegexMatcher`